x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/amazonbedrock/AmazonBedrockInferenceClient.java

```diff
@@ -0,0 +1,165 @@
+            throw elasticsearchException;
+        } catch (Exception e) {
+            throw new ElasticsearchException("Amazon Bedrock client converse call failed", e);
+        }
+    }
+
+    @Override
+    public InvokeModelResult invokeModel(InvokeModelRequest invokeModelRequest) throws ElasticsearchException {
+        try {
+            return internalClient.invokeModel(invokeModelRequest);
+        } catch (AmazonBedrockRuntimeException amazonBedrockRuntimeException) {
+            throw new ElasticsearchException(
+                Strings.format("failed to create AmazonBedrockRuntime client: [%s]", amazonBedrockRuntimeException.getMessage()),
+                amazonBedrockRuntimeException
+            );
+        } catch (ElasticsearchException elasticsearchException) {
+            // just throw the exception if we have one
+            throw elasticsearchException;
+        } catch (Exception e) {
+            throw new ElasticsearchException("Amazon Bedrock client invokeModel call failed", e);
+        }
+    }
+
+    protected AmazonBedrockRuntime createAmazonBedrockClient(AmazonBedrockModel model, @Nullable TimeValue timeout) {
+        var secretSettings = (AmazonBedrockSecretSettings) model.getSecretSettings();
+        var credentials = new BasicAWSCredentials(secretSettings.accessKey.toString(), secretSettings.secretKey.toString());
+        var credentialsProvider = new AWSStaticCredentialsProvider(credentials);
+        var clientConfig = timeout == null
+            ? new ClientConfiguration().withConnectionTimeout(DEFAULT_CLIENT_TIMEOUT_MS)
+            : new ClientConfiguration().withConnectionTimeout((int) timeout.millis());
+
+        var serviceSettings = (AmazonBedrockServiceSettings) model.getServiceSettings();
+
+        try {
+            SpecialPermission.check();
+            AmazonBedrockRuntimeClientBuilder builder = AccessController.doPrivileged(
+                (PrivilegedExceptionAction<AmazonBedrockRuntimeClientBuilder>) () -> AmazonBedrockRuntimeClientBuilder.standard()
+                    .withCredentials(credentialsProvider)
+                    .withRegion(serviceSettings.region())
+                    .withClientConfiguration(clientConfig)
+            );
+
+            return SocketAccess.doPrivileged(builder::build);
+        } catch (AmazonBedrockRuntimeException amazonBedrockRuntimeException) {
+            throw new ElasticsearchException(
+                Strings.format("failed to create AmazonBedrockRuntime client: [%s]", amazonBedrockRuntimeException.getMessage()),
+                amazonBedrockRuntimeException
+            );
+        } catch (Exception e) {
+            throw new ElasticsearchException("failed to create AmazonBedrockRuntime client", e);
+        }
+    }
+
+    private void setExpiryTimestamp() {
+        synchronized (this) {
+            this.expiryTimestamp = System.currentTimeMillis() + CACHE_EXPIRY_ADD_MS;
+        }
+    }
```

Comment:
```suggestion
    private synchronized void setExpiryTimestamp() {
            this.expiryTimestamp = System.currentTimeMillis() + CACHE_EXPIRY_ADD_MS;
    }
```

nit: is there a difference between `synchronized` on the method level and `synchronized(this)?`, if all of the code resides inside `synchronized(this) {}`. If not I think we can synchronize on the method level, then it's more obvious to the caller that everything is synchronized

Reply:
Ah - I'll doublecheck... I think I may be conflating some thread-safe mechanisms from my C# days...